Automattic/sensei
# Add scheduling and sync

Fixes #6430 
Based on #6485

### Changes proposed in this Pull Request

* Add cron task to run syncing daily.
* Add hooks to update subscribers when students are added to a course or group.
* Add hooks to update subscribers when students are removed from a course or group.

### Testing instructions

- ~Cron~
  - ~After one page load, use WP CLI to run the following command: `wp cron event list --fields=hook,next_run`~
  - ~Check to see that the task `sensei_sync_mailpoet_subscribers` is scheduled to run.~
  - ~Alternatively use a cron management plugin to see scheduled tasks.~
- Hooks
  - Enrol a student to a course 
  - Check the MailPoet List page to see it added as a subscriber. MailPoet > Subscribers, select the course on the drop down filter and see the student added with status 'Unconfirmed'.
  - Remove the students' enrolment and check back in MailPoet to see status 'Unsubscribed'.
  - Repeat the same by adding and removing students from any group.

Update: Replaced `wp-cron` implementation with `SenseiScheduler`
